explainer.md

```diff
@@ -68,356 +107,405 @@ If the set of pages being badged corresponds to an [Installed Web Application](h
-  void clear(optional BadgeOptions options);
-}
-```
+* Can be used for much higher frequency / lower priority events than notifications, because each new event does not disrupt the user.
+* There may be no need to request permission to use the badging API, since it is much less invasive than a notification.
 
-> Note: Should we have a separate overload for boolean flags now, as discussed in [Issue 19](https://github.com/WICG/badging/issues/19) and [Issue 42](https://github.com/WICG/badging/issues/42)?
+Typically, sites will want to use both APIs together: notifications for high-importance events such as new direct messages or incoming calls, and badges for all new messages including group chats not directly addressed to the user.
 
-> Note: This API can only be used from a foreground page. Use from a service worker is being considered for the future. [The FAQ](#Why-cant-this-be-used-in-the-background-from-the-ServiceWorker-see-28-and-5) has more details).
+## Badge scope
 
-### Examples
+This section explains the difference between badging a *set of URLs* versus
+badging a *document*.
 
-#### Basic Examples
-These examples provided limited context. They are intended to show how the API will look.
+Consider four increasingly narrow sets of pages that a Badge is applied to:
 
-##### Setting an integer badge (as in an email app):
-```js
-Badge.set(getUnreadCount());
-```
+1. An entire origin (e.g., when you just want to set an unread count for the
+   entire site).
+2. All pages whose URL [starts
+   with](https://www.w3.org/TR/appmanifest/#dfn-within-scope) a given path
+   (e.g., if a site is organised into "projects", each with their own status
+   indicator).
+3. All pages with a specific URL path (e.g., if each page has a separate status
+   indicator, but all tabs open at a certain URL should share status).
+4. A specific document (e.g., if many tabs can be open with URLs on the same
+   path, but might have different status, due to either a different query
+   string, or different dynamic context inside the document).
 
-##### Setting and clearing a boolean flag (as in a game of chess):
-```js
-if (myTurn())
-  Badge.set();
-else
-  Badge.clear();
-```
+Number 3 and 4 seem very similar, but architecturally are very different. In
+3, the badge is associated with a URL (and any current or future documents open
+at that URL will show that badge), whereas in 4, the badge is associated with a
+particular document — other tabs open to the same URL will not share the badge.
 
-#### Advanced Examples
-These examples generally provide more context around what is going on, and may specify scopes. They are intended to show how real world applications may use the API.
+What this boils down to is two different modes:
 
-##### Updating a Badge on a message from a WebSocket (as in a messaging app, receiving new messages):
+* Badging a particular URL scope, applied to the set of URLs that start with the
+  same path. This covers 1, 2 and 3 above.
```

Comment:
Just *barely*, if you *squint* `3`.

Reply:
Why? Do you mean because of the prefix match thing? Hmm.

I suppose the real problem here is that some "endpoint" paths will not end in a slash, for example "/users/matt", so if you want a per-user badge, you'll have to badge "/users/matt" and NOT "/users/matt/". But that means you're also badging "/users/matthew" and anything else that starts with that prefix.

.... more complexity.

Reply:
I'm more about the issue where you display a list of users at `/users/` and badge that with a badge which should **only** apply to that page. It'd be kind of weird if it also showed up for `/users/matt`.

Another example:

Builds at `/build/<number>` (check or cross for success/fail)
Build logs at `/builds/<number>/logs` (plain text, shouldn't really have a badge).

Reply:
Right. So we don't have a way to badge a single URL. But I guess this all goes away if we split the API. The scoped badging is for broad groups of URLs (i.e., an entire app) rather than individual pages.

Reply:
:+1: